elf_reader.go

```diff
@@ -533,7 +533,21 @@ func (ec *elfCode) loadBTFMaps(maps map[string]*MapSpec) error {
 				return fmt.Errorf("section %v: map %v already exists", sec.Name, sym)
 			}
 
-			mapSpec, err := mapSpecFromBTF(ec.btf, name)
+			// A global Var is created by declaring a struct with a 'structure variable',
```

Comment:
This can happen in a follow up, ultimately I think we should iterate the Datasec directly.

```
$ bpftool btf dump file testdata/loader-clang-11-el.elf
...
[12] VAR 'hash_map' type_id=11, linkage=global-alloc
[20] VAR 'hash_map2' type_id=19, linkage=global-alloc
[22] VAR 'btf_pin' type_id=21, linkage=global-alloc
[52] DATASEC '.maps' size=0 vlen=3
	type_id=12 offset=0 size=40
	type_id=20 offset=0 size=32
	type_id=22 offset=0 size=40
```

So instead of looping over sec.symbols try to `FindType(secName, &btf.Datasec)` and then iterate Datasec.Vars. That gives us access to VarSecinfo.Offset and VarSecinfo.Size which we'll need when trying to match relocations against the maps section.

Aside: kind of weird that `offset=0` in the dump, I wonder what's up with that.

Reply:
Spun this off into https://github.com/cilium/ebpf/issues/239.